### PR TITLE
chore(deps): Migrate dependency management to use `poetry`

### PR DESCRIPTION
**Pull request by setu4993**
Switching over dependency management to `poetry`!

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,130 +1,121 @@
...
+setfit = {version ="==0.7.0", optional = true}
+
+# Cuda.
+ucx-py-cu11 = { version ="^0.36", source = "nvidia", optional = true }
+rmm-cu11 = {version = "^24.2.0", source = "nvidia", optional=true}
+raft-dask-cu11 = {version = "^24.2.0", source = "nvidia", optional=true}
+pylibraft-cu11 = {version = "^24.2.0", source = "nvidia", optional=true}
+dask-cudf-cu11 = {version = "^24.2.0", source = "nvidia", optional=true}
+cudf-cu11 = {version = "^24.2.0", source = "nvidia", optional=true}
+cuml-cu11 = {version = "^24.2.0", source = "nvidia", optional=true}
```
Comment (elboy3) [inline pyproject.toml:60]: Would be nice to have these as an optional extra, is that possible with poetry?

**Review by elboy3 — COMMENTED** [review]
Looking really good, heroic stuff setu! Your PR description says WIP, what's left?

**Comment by setu4993** [comment]
@elboy3 :

> Looking really good, heroic stuff setu! Your PR description says WIP, what's left?

Yup, it was WIP when I first created it. What I'm still waiting for is confirmation that we can drop Python 3.8 support.

**Comment by elboy3** [comment]
@setu4993 sounds good. We have confirmation from Legartis, Reddit, and Musixmatch thus far. 

I will confirm with the remaining customers after my 1-1 with yash today

**Comment by elboy3** [comment]
@setu4993 sounds good, will look into that later today

**Comment by franz101** [comment]
How did you solve the Cuda issues?

**Comment by setu4993** [comment]
> How did you solve the Cuda issues?

Used the latest versions :).

File: LICENSE [hunk 2]
```diff
@@ -1,7 +1,201 @@
-Copyright 2021 Galileo Technologies, Inc.
+                                 Apache License
```
Comment (setu4993) [inline LICENSE:1]: Our license was MIT, which doesn't seem correct for what we are supporting. It doesn't even offer trademark protections!

File: pyproject.toml [hunk 3]
```diff
@@ -1,131 +1,122 @@
...
-]
-minio = ["minio>=7.1.0,<7.2.0"]
-setfit = ["setfit==0.7.0"]
+license = "Apache-2.0"
+homepage = "https://github.com/rungalileo/dataquality"
+documentation = "https://rungalileo.gitbook.io/galileo"
+scripts = { dqyolo = "dataquality.dqyolo:main" }
+
+[tool.poetry.dependencies]
+python = "^3.9,<3.12"
```
Comment (elboy3) [inline pyproject.toml:17]: shouldn't this be <3.11. i don't think vaex and 3.11 are compatible yet

File: .github/workflows/test.yaml [hunk 4]
```diff
@@ -13,34 +13,32 @@ jobs:
...
           python-version: ${{ matrix.python-version }}
+          cache-dependency-path: "pyproject.toml"
 
       - name: install invoke
-        run: pip install invoke
+        run: pipx install invoke
 
       - name: install dependencies
-        run: inv install --no-editable
+        run: invoke install
```
Comment (elboy3) [inline .github/workflows/test.yaml:36]: why this?

**Review by elboy3 — APPROVED** [review]
Looks great except i think we should also just restrict 3.9 and 3.10 in poetry file

File: .github/workflows/test.yaml (hunk 4 above)
Comment (setu4993) [reply .github/workflows/test.yaml:36]: I removed the conditional from `tasks.py`, so this makes it more aligned.

**Pull request merged**